tfc-dump.pl

```diff
@@ -19,6 +19,34 @@
 use strict;
 use warnings;
 use Getopt::Long qw(GetOptions);
+use File::Slurp qw(read_file);
+use File::Spec;
+
+# Function to log errors to Sentry
+sub error_to_sentry {
+    my ($message) = @_;
+    system("sentry-cli send-event \"$message\" 2>/dev/null") if $ENV{SENTRY_DSN};
+}
+
+# slurp output function
+sub slurp_output {
+    my ($command, $filename) = @_;
+    
+    my $tmp_file = File::Spec->catfile('/tmp', $filename);
+    
+    system("$command > $tmp_file");
+    if ($?) {
+        my $error = "Failed to execute command: $command";
+        error_to_sentry($error);
+        die "$error\n";
+    }
+    
+    my $content = read_file($tmp_file);
+    unlink($tmp_file);
+    
+    chomp($content);
+    return $content;
+}
```

Comment:
Much nicer. :)

Reply:
If you extract 44-48 to a get_string_from($tmp_file) and write a get_array_from($tmp_file), then you could expand slurp_output to return either an array or string...